src/commands/dataconnect-sdk-generate.ts

```diff
@@ -20,36 +21,42 @@ export const command = new Command("dataconnect:sdk:generate")
     const projectId = needProjectId(options);
 
     const serviceInfos = await loadAll(projectId, options.config);
-    for (const serviceInfo of serviceInfos) {
-      const configDir = serviceInfo.sourceDirectory;
-      const hasGeneratables = serviceInfo.connectorInfo.some((c) => {
+    const serviceInfosWithSDKs = serviceInfos.filter((serviceInfo) =>
+      serviceInfo.connectorInfo.some((c) => {
         return (
           c.connectorYaml.generate?.javascriptSdk ||
           c.connectorYaml.generate?.kotlinSdk ||
           c.connectorYaml.generate?.swiftSdk ||
           c.connectorYaml.generate?.dartSdk
         );
+      }),
+    );
+    if (!serviceInfosWithSDKs.length) {
+      logger.warn("No generated SDKs have been declared in connector.yaml files.");
+      logger.warn(`Run ${clc.bold("firebase init dataconnect:sdk")} to configure a generated SDK.`);
+      logger.warn(
+        `See https://firebase.google.com/docs/data-connect/web-sdk for more details of how to configure generated SDKs.`,
+      );
+      return;
+    }
+    const generateRuns = serviceInfosWithSDKs.map((serviceInfo) => {
+      const configDir = serviceInfo.sourceDirectory;
+      const account = getProjectDefaultAccount(options.projectRoot);
+      return DataConnectEmulator.generate({
+        configDir,
+        watch: options.watch,
+        account,
       });
-      if (!hasGeneratables) {
-        logger.warn("No generated SDKs have been declared in connector.yaml files.");
-        logger.warn(
-          `Run ${clc.bold("firebase init dataconnect:sdk")} to configure a generated SDK.`,
-        );
-        logger.warn(
-          `See https://firebase.google.com/docs/data-connect/web-sdk for more details of how to configure generated SDKs.`,
-        );
-        return;
-      }
-      for (const conn of serviceInfo.connectorInfo) {
-        const account = getProjectDefaultAccount(options.projectRoot);
-        const output = await DataConnectEmulator.generate({
-          configDir,
-          connectorId: conn.connectorYaml.connectorId,
-          watch: options.watch,
-          account,
-        });
-        logger.info(output);
-        logger.info(`Generated SDKs for ${conn.connectorYaml.connectorId}`);
-      }
+    });
+
+    if (options.watch) {
+      await Promise.race(generateRuns);
+    } else {
+      await Promise.all(generateRuns);
```

Comment:
This runs them all at once and may cause the stdout to interleave each other. The previous code containing for-loop works a lot better here.

Reply:
Okay, sure. Did the refactor to make it serial.